### ISSUE TITLE
indent awk does not work correctly

### ISSUE THREAD

**Issue by jiangyinzuo** [vim/vim#13201]
### Steps to reproduce

vim --clean foo.awk
```awk
{
	if ($1 == "x") {
					print $1
	} else    {
	  print $2
}
}

```

### Expected behaviour

```awk
{
        if ($1 == "x") {
                print $1
        } else    {
                print $2
        }
}
```

but actually
```awk
{
	if ($1 == "x") {
		print $1
	} else    {
	print $2
}
}

```

### Version of Vim

VIM - Vi IMproved 9.0 (2022 Jun 28, compiled May 24 2023 14:26:32) Included patches: 1-1000, 1087, 1117-1118, 1129, 1165, 1143-1145, 1189

### Environment

Ubuntu 23.04

**Comment by jiangyinzuo** [comment]
By the way, I found a simple plugin [indent-awk.vim](https://github.com/Clavelito/indent-awk.vim) that works correctly.

**Comment by chrisbra** [comment]
can you please contact the author of the indent script: https://github.com/vim/vim/blob/master/runtime/indent/awk.vim#L19

**Comment by chrisbra** [comment]
the question is, what does regress in that case 🤷 ?

**Comment by chrisbra** [comment]
Oh, I see that is also in vim9 script. not sure if this could potentially cause issues for Neovim.

**Comment by clason** [comment]
Indeed, we wouldn't be able to include it and would have to freeze the indent script at the moment.

**Comment by dkearns** [comment]
@clason, what is your policy regarding direct runtime submissions?  If someone were to hypothetically port this to legacy script would it be accepted or is the divergence too much of a burden?

**Comment by clason** [comment]
Our policy right now is a hard "no"; we instantly direct such submissions to the proper channels (contact maintainer at the top of the file). So far, this hasn't changed for files that have been ported to vim9script:
Either
1. the current transpiler (which is still limited in capability, even for syntax is already relatively stable) can handle it,  then we use that and include the transpiled file;
2. if the file is just a straight 1:1 rewrite, we can usually make equivalent changes manually (which is not great fun, I have to admit);
3.  otherwise we freeze them. So far, nobody has tried to do "bonus work" on those...

So there would be precedent at least, if this hypothetical port fell into category 2. At some point though, we'd have to come to a decision (which we have discussed before, of course -- the writing's been on the wall for a while, after all -- but always punted on so far) how much divergence makes sense from a maintenance point of view.
